kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala

```diff
@@ -823,13 +823,21 @@ object KyuubiConf {
       .version("1.4.0")
       .fallbackConf(ENGINE_SHARE_LEVEL_SUB_DOMAIN)
 
+  @deprecated(s"using ${FRONTEND_CONNECTION_URL_USE_HOSTNAME.key} instead, 1.5.0")
   val ENGINE_CONNECTION_URL_USE_HOSTNAME: ConfigEntry[Boolean] =
     buildConf("engine.connection.url.use.hostname")
-      .doc("When true, engine register with hostname to zookeeper. When spark run on k8s" +
+      .doc("(deprecated) " +
+        "When true, engine register with hostname to zookeeper. When spark run on k8s" +
         " with cluster mode, set to false to ensure that server can connect to engine")
       .version("1.3.0")
       .booleanConf
-      .createWithDefault(true)
+      .createWithDefault(false)
+
+  val FRONTEND_CONNECTION_URL_USE_HOSTNAME: ConfigEntry[Boolean] =
+    buildConf("frontend.connection.url.use.hostname")
```

Comment:
I think it's `backend` ? the original idea of this config is used for that spark engine register its hostname or ipaddress to zk.

Reply:
it's Frontend

Reply:
ah, I see. This PR refactor the related code and unify the connect url. But it is actually used by both server and engine. seems both `frontend` and `backend ` are not good.

Reply:
this conf will be propagated to other frontends and connection URL is part of base FE trait